test/torchaudio_unittest/prototype/functional/functional_test_impl.py

```diff
@@ -518,3 +522,294 @@ def _debug_plot():
         except AssertionError:
             _debug_plot()
             raise
+
+
+class FunctionalCPUOnlyTestImpl(TestBaseMixin):
+    @parameterized.expand(
+        [
+            (0.1, 0.2, (2, 1, 2500)),  # both float
+            # Per-wall
+            (torch.rand(4), 0.2, (2, 1, 2500)),
+            (0.1, torch.rand(4), (2, 1, 2500)),
+            (torch.rand(4), torch.rand(4), (2, 1, 2500)),
+            # Per-band and per-wall
+            (torch.rand(6, 4), 0.2, (2, 6, 2500)),
```

Comment:
The shape of `e_absorption` doesn't seem correct. In multi-band case we assume the number of octave bands must be 7, right?

Reply:
We do not need this ``num_bands == 7`` assumption for ray tracing so we can be more generic. I can restrict the number of bands to 7 if you prefer, LMK. See also https://github.com/pytorch/audio/pull/2850#discussion_r1040696715